test/lit/exec/table64.wast

```diff
@@ -30,6 +30,16 @@
   )
  )
 
+ ;; CHECK:      [fuzz-exec] calling oob-huge
+ ;; CHECK-NEXT: [trap callTable overflow]
+ (func $oob-huge (export "oob-huge") (result i32)
```

Comment:
It might be worth it to place a function in the table such that if we did incorrectly wrap the address, it would call the function and not trap at all, rather than trap with the wrong error message. I can imagine that if we rewrote or refactored the interpreter, we could easily miss this problem otherwise.

Reply:
There are call_indirect tests, but it looks like they don't check for wrapping. I'll add some.

Reply:
@sbc100 it doesn't look like the call_indirect tests have been updated to use i64 indices for 64-bit tables: https://github.com/WebAssembly/memory64/blob/main/test/core/call_indirect.wast#L164-L166. Am I reading that correctly?

Reply:
I see the following changes to that file which do includes two calls to call_indirect with i64 index:

```
diff --git a/test/core/call_indirect.wast b/test/core/call_indirect.wast
index 79b8dc39..ba267fea 100644
--- a/test/core/call_indirect.wast
+++ b/test/core/call_indirect.wast
@@ -61,10 +61,15 @@
     )
   )
 
+  (table $t64 i64 funcref
+    (elem $const-i32)
+  )
+
   ;; Syntax
 
   (func
     (call_indirect (i32.const 0))
+    (call_indirect $t64 (i64.const 0))
     (call_indirect (param i64) (i64.const 0) (i32.const 0))
     (call_indirect (param i64) (param) (param f64 i32 i64)
       (i64.const 0) (f64.const 0) (i32.const 0) (i64.const 0) (i32.const 0)
@@ -94,6 +99,9 @@
   (func (export "type-i32") (result i32)
     (call_indirect (type $out-i32) (i32.const 0))
   )
+  (func (export "type-i32-t64") (result i32)
+    (call_indirect $t64 (type $out-i32) (i64.const 0))
+  )
   (func (export "type-i64") (result i64)
     (call_indirect (type $out-i64) (i32.const 1))
   )
@@ -474,6 +482,8 @@
 (assert_return (invoke "type-f64") (f64.const 0xf64))
 (assert_return (invoke "type-f64-i32") (f64.const 0xf64) (i32.const 32))
 
+(assert_return (invoke "type-i32-t64") (i32.const 0x132))
+
 (assert_return (invoke "type-index") (i64.const 100))
 
 (assert_return (invoke "type-first-i32") (i32.const 32))

```